Gems/LmbrCentral/Code/Source/Shape/CylinderShape.cpp

```diff
@@ -260,10 +259,58 @@ namespace LmbrCentral
             AZ::Vector3 diff = m_intersectionDataCache.m_baseCenterPoint - point;
             return diff.GetLengthSq();
         }
-        return Distance::Point_CylinderSq(
-            point, m_intersectionDataCache.m_baseCenterPoint,
-            m_intersectionDataCache.m_baseCenterPoint + m_intersectionDataCache.m_axisVector,
-            m_intersectionDataCache.m_radius);
+        // Use the cylinder axis' center point to determine distance by
+        // splitting into Voronoi regions and using symmetry.
+        // The regions are:
+        // - Inside
+        // - Beyond cylinder radius but between two disc ends.
+        // - Within cylinder radius but beyond two disc ends.
+        // - Beyond cylinder radius and beyond two disc ends.
+        float radius = m_intersectionDataCache.m_radius;
```

Comment:
As this was a method available previously shouldn't we add that in az intersection ?

Reply:
Is there any other users for the legacy function beside this place?

Reply:
> Is there any other users for the legacy function beside this place?

One issue is because of the distributed nature of O3DE, we don't know the answer to that... code can live in what user gems exist, or things in o3de-extras, or what others are using in their private games and plugins.
   
When we remove APIS we like to, if possible, leave a deprecation notice, and if we can resolve the old code to still compile via a `using oldtype = newtype` or something, even better.